// atom/index.js
var gulp = require('gulp');
var EventEmitter = require('events').EventEmitter;

var Atom = function(){
    this.parsers = [];
    var event = new EventEmitter();
    this.paths = [];
    var self = this;
    this.locations = {
      'elements':'elements/**/*.html'  
    };
    this.createLocations = function(name){
        var loc = this.locations[name];
        var res = [];
        this.paths.forEach(function(value){
            res.push(value+'/'+loc);
        });
    }
    this.setParser = function(selector,func,setting){
        this.parsers.push({
            element:selector,
            func:func,
            settings:setting
        });
    };
    this.setPath = function(path){
        this.paths.push(path);
    };
    this.stripFile = function(text){
        
    }
    this.defineBaseTasks  = function(e){
        event.addListener('taskDefinition',function(){
            
        });
        event.addListener('watch',function(){
            gulp.watch('elements/**/*.html','stripe');
        });
    }
    this.run = function(){
        this.defineBaseTasks();
        this.parsers.forEach(function(value,index){
            value.func(event);
        })
        gulp.task('watch',function(){
          event.emit('watch');
        });
        
    }
    this.watch = function(){
        gulp.start('watch');
    }
};

module.exports = new Atom();
